sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/ByteExtensions.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Security.KeyVault.Keys.Cryptography
+{
+    internal static class ByteExtensions
+    {
+        internal static bool SequenceEqualConstantTime(this byte[] self, byte[] other)
+        {
+            if (self == null)
+                throw new ArgumentNullException(nameof(self));
+
+            if (other == null)
+                throw new ArgumentNullException(nameof(other));
+
+            // Constant time comparison of two byte arrays
+            uint difference = (uint)self.Length ^ (uint)other.Length;
```

Comment:
Why must this be constant time? We could short-circuit these checks for efficiency.

Reply:
Comparisons in cryptography should be constant time to prevent timing based attacks.